tests/test_trie.py

```diff
@@ -22,3 +22,29 @@ def test_trie():
 
     trie.add("fo")
     assert trie.pattern() == "(?:ab(?:c|s(?:olute)?|xy?)|fo[eo]?)"
+
+    trie = Trie()
+    trie += Trie("abc")
+    assert trie.pattern() == "abc"
+    assert (
+        trie + Trie().add("foo")
+        == Trie("abc", "foo")
+        == Trie(["abc", "foo"])
+        == Trie().add(["abc", "foo"])
+        == Trie().add("abc", "foo")
+        == Trie().add("abc").add("foo")
+    )
+
+    try:
+        trie + None
+    except TypeError:
+        pass
+    else:
+        assert False, "TypeError not raised"
```

Comment:
for the failing flake8 you can use [`pytest.raises`](https://github.com/ddelange/mapply/blob/1762038c1f66c1f88ce5cf06fac852d6516c0e4f/tests/test_mapply.py#L145-L146)